### PR TITLE
Fix parsing remote sampling strategy on Python 3.5 (#180)

### LINKED ISSUE #180: Jaeger client can't parse remote sampling strategy with Python 3.5
## Requirement - what kind of business use case are you trying to solve?

Run Jaeger Python Client with Python 3.5

## Problem - what in Jaeger blocks you from solving the requirement?

Python 3.5 complains about not being able to load JSON with sampling strategy from agent since it's not str but bytes.

```
Fail to parse sampling strategy from jaeger-agent: the JSON object must be str, not 'bytes' [b'{"strategyType":"PROBABILISTIC","probabilisticSampling":{"samplingRate":1}}']
```

## Proposal - what do you suggest to solve the problem or improve the existing situation?

In sampler.py decode `response.body` to str:
```diff
-            sampling_strategies_response = json.loads(response.body)
+            sampling_strategies_response = json.loads(response.body.decode('utf-8'))
```
https://github.com/jaegertracing/jaeger-client-python/blob/master/jaeger_client/sampler.py#L417

## Any open questions to address

None

### PR DESCRIPTION
**Pull request by nielspardon**
Fixes #180

### CODE REVIEW

File: tests/test_sampler.py [hunk 1]
```diff
@@ -419,6 +419,88 @@ def test_sampling_request_callback():
 
     sampler.close()
 
+# noinspection PyProtectedMember
+def test_sampling_request_callback_bytes():
```
Comment (yurishkuro) [inline tests/test_sampler.py]: this seems like an awfully long test just to check the return type. Maybe one of the existing tests can be extended instead?

File: tests/test_sampler.py [hunk 2]
```diff
@@ -400,6 +400,28 @@ def test_sampling_request_callback():
...
+    assert '%s' % sampler.sampler == 'ProbabilisticSampler(0.002)', 'sampler should have changed to probabilistic'
+    prev_sampler = sampler.sampler
+
+    sampler._sampling_request_callback(return_value)
+    assert prev_sampler is sampler.sampler, "strategy hasn't changed so sampler should not change"
+
+    adaptive_sampling_strategy_bytearray = bytearray(adaptive_sampling_strategy.encode('utf-8'))
+
+    return_value.result = lambda *args: \
+        type('obj', (object,), {'body': bytearray(adaptive_sampling_strategy_bytearray)})()
```
[the remark this answers is not in the document]
Comment (nielspardon) [reply tests/test_sampler.py]: good catch, will fix that :-)

File: tests/test_sampler.py [hunk 3]
```diff
@@ -412,6 +434,18 @@ def test_sampling_request_callback():
...
     assert prev_sampler is sampler.sampler, 'error updating sampler should not update the sampler'
 
+    return_value.result = lambda *args: \
+        type('obj', (object,), {'body': None})()
+    sampler._sampling_request_callback(return_value)
+    assert error_reporter.error.call_count == 3
+    assert prev_sampler is sampler.sampler, 'error updating sampler should not update the sampler'
+
+    return_value.result = lambda *args: \
+        type('obj', (object,), {'body': {'encode': None}})()
```
[the remark this answers is not in the document]
Comment (nielspardon) [reply tests/test_sampler.py]: yes, good catch, I accidentally had a similar typo earlier, will fix this one

File: tests/test_sampler.py [hunk 4]
```diff
@@ -400,6 +400,28 @@ def test_sampling_request_callback():
...
+    adaptive_sampling_strategy_bytearray = bytearray(adaptive_sampling_strategy.encode('utf-8'))
+
+    return_value.result = lambda *args: \
+        type('obj', (object,), {'body': bytearray(adaptive_sampling_strategy_bytearray)})()
+    sampler._sampling_request_callback(return_value)
+    assert '%s' % sampler.sampler == 'AdaptiveSampler(0.001000, 2.000000, 10)', 'sampler should have changed to adaptive'
+    prev_sampler = sampler.sampler
+
+    sampler._sampling_request_callback(return_value)
+    assert prev_sampler is sampler.sampler, "strategy hasn't changed so sampler should not change"
```
[the remark this answers is not in the document]
Comment (nielspardon) [reply tests/test_sampler.py]: good, will drop that

[the remark this answers is not in the document]
Comment (nielspardon) [reply tests/test_sampler.py]: I still need to keep line 420 for the test following this one. Just in case you are wondering.

**Comment by nielspardon** [comment]
I applied all requested changes.

**Comment by yurishkuro** [comment]
Thanks for the fix!

**Pull request merged**